### ISSUE TITLE
BorderLayout:ed panel resize doesn't re-layout

### ISSUE THREAD

**Issue by mabe02** [mabe02/lanterna#166]
See the window manager test, when opening the full screen window and then making the terminal window bigger, the component doesn't resize.

**Comment by mabe02** [comment]
This is now fixed; Window will call invalidate() on the main component whenever the size has changed

**Comment by avl42** [comment]
That one seems fixed, but there is another little thing... (solved below)

Before I saw (and pulled) your fix, I also had a look at it, and tried adding a Border around the TextBox to better see its actual size.  Strangely enough, as soon as I added the Border, the component disappeared completely. And that didn't even change with your fix!  I looked more, and noticed, that the BorderLayout sets all components to zero-size that don't have a Location as their LayoutData!

It seems that without the Border, the textbox was set to zero size but it ignored it, whereas the Border felt obliged by it, and furtheron didn't draw anything (nor it's content)

So, the "bug" was, that on adding the textbox to a BorderLayout, you forgot to tell it the location.
But then again, I think this will likely trick people again in future, and apparently it may remain hidden for quite some time, so either you should really throw an exception for lacking Location, or (what I'd prefer) make CENTER the default, and only (but then for sure) throw an exception, if two components would end up in same location.

(This time I do it like in the old days on googlecode, posting the snippet here)
here is my suggested replacement for the loop-body of BorderLayout.makeLookupMap:
```java
            LayoutData ld = component.getLayoutData();
            if( ld == null ) { ld = Location.CENTER; }
            if( ld instanceof Location) {
                if ( map.put((Location)ld, component) != null ) {
                    throw new IllegalArgumentException("Multiple components set for "+ld+" location.");
                }
            }
```
Feel free to change code style, variable names and the exact exception type as you please.

**Comment by mabe02** [comment]
Yeah, I don't want to make BorderLayout too complicated; let's assign components without any layout data to an available location and if all locations are taken then throw.

**Comment by mabe02** [comment]
Well, maybe throwing is a bit harsh. I've updated it to auto-assign if it can, otherwise the components will be set to size 0

**Comment by avl42** [comment]
I looked at the diff and it looks fine.

By not throwing an Exception, however it still leaves a possibility that some component might end up in this strange state where it's visibility depends on not having a Border. Even though this situation should be quite rare now (would require six components added to a single BL), I don't want to forget about it yet.

I noticed that I don't even understand how the original TextBox (without Border and sized to zero) could even show up in the first place.  In case I find out more, I'll probably open a new Issue, as that will (most likely) not be BorderLayout-related.
